app/src/main/java/ch/epfl/sweng/eventmanager/repository/EventRepository.java

```diff
@@ -2,62 +2,67 @@
 
 import android.arch.lifecycle.LiveData;
 import android.arch.lifecycle.MutableLiveData;
+import android.support.annotation.NonNull;
 import ch.epfl.sweng.eventmanager.repository.data.Event;
-import ch.epfl.sweng.eventmanager.repository.data.EventOrganizer;
+import com.google.firebase.database.*;
 
 import javax.inject.Inject;
 import javax.inject.Singleton;
-import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
 
 /**
  * @author Louis Vialar
  */
 @Singleton
 public class EventRepository {
-    // In the future, the events data will come from Firebase
-    // From now, we just get them from a static list
-    private final List<Event> EVENTS = new ArrayList<>(3);
-
-    {
-        EVENTS.add(
-                new Event(1, "Japan Impact", "La plus grande convention sur la culture japonaise " +
-                        "de Suisse Romande !",
-                        new EventOrganizer(1, "PolyJapan", "La commission de l'AGEPoly qui promeut la culture japonaise sur le campus et ses environs", null),
-                        null));
-        EVENTS.add(
-                new Event(2, "Sysmic", "Le festival de musique de l'association des étudiants de Microtechnique",
-                        new EventOrganizer(2, "Sysmic", "L'association des étudiants de Microtechnique", null),
-                        null));
-        EVENTS.add(
-                new Event(3, "Souper de section", "Retrouve tes professeurs et camarades lors du souper de section",
-                        new EventOrganizer(3, "CLIC", "L'association des étudiants en IC", null),
-                        null));
-    }
 
     @Inject
-    public EventRepository() {}
+    public EventRepository() {
+    }
 
     public LiveData<List<Event>> getEvents() {
         final MutableLiveData<List<Event>> data = new MutableLiveData<>();
+        DatabaseReference dbRef = FirebaseDatabase.getInstance().getReference("events");
 
-        data.setValue(Collections.unmodifiableList(EVENTS));
+        dbRef.addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                GenericTypeIndicator<List<Event>> typeToken = new GenericTypeIndicator<List<Event>>() {
+                };
+
+                List<Event> events = dataSnapshot.getValue(typeToken);
+                if (events != null)
+                    events.remove(null); // Somehow the list might contain a null element
```

Comment:
There is most likely something wrong on the stored data. https://stackoverflow.com/questions/15534917/why-do-firebase-collections-seem-to-begin-with-a-null-row looks similar from the firebase point of view.

If we can't fix this right now I'd like to mark it with a `FIXME` tag at the beginning of the comment, in order to make it easier to find/more visible.

Reply:
Yep.
I uploaded the data as a `Map<String, Event>` where the key is `String.valueOf(event.getId());`, but firebase sees it as a `List<Event>` (we should prefix the names of the keys to get a map). As the IDs start at 1, the list's first element is null. 

We have therefore three options to fix this: 
 - Make a proper map by prefixing the keys with "event-" 
 - Add an element with id 0
 - Keep this code and consider it normal

I think the first solution is the best, I let you judge.

Reply:
In that case, wouldn't it make sense to use a `Map<Int, Event>` ?

Reply:
Nope, maps are required to have a `String` as a key in Firebase. 
And I first tried to pull the data as a `Map<String, Event>` but Firebase was passing me a `List` anyway

Reply:
Can't we use a simple list then? I don't like weird frankenstein keys.

Reply:
Yes, but then we lose the ability to query a single event by its id (or we must take for granted the fact that key == id, which we can enforce, but might generate ranges of empty (and therefore null) values in the list

Reply:
I don't like the idea of concatenating strings to get the key of my object. In any case, the first two solutions won't stop me from sleeping as long as it's properly documented. I let the choice to your discretion.